Boltdb TestOpenDatabase function

mnsl opened the pull request:
Implements new TestOpenDatabase function that tests the OpenDatabase function and BoltDatabase Close method in package persist.  Checks cases where database is empty, full, and has not yet been created.

DavidVorick commented on persist/boltdb_test.go:
```diff
@@ -0,0 +1,167 @@
+	[]byte("FakeBucket" + RandomSuffix()),
+	[]byte("_"),
+	[]byte(" asdf"),
+}
```
Since these variables and costs are only used inside of one test, it's better to declare them inside that test, especially since they don't really have broader application.

lukechampine replied:
Also, note that you can inline the struct definition:
```go
testInputs := make([]struct{
    meta Metadata
    filename string
}, numTestInputs)
```

DavidVorick commented on persist/boltdb_test.go:
```diff
@@ -0,0 +1,167 @@
+		t.Fatal(err)
+	}
+
+	// Generate ten random testInputs for building database files.
```
better not to use a word like 'ten', because sometimes you change the variable but forget the constant :) (same problem you fixed above - probably better to remove the number altogether here too)

lukechampine commented on persist/boltdb_test.go:
```diff
@@ -0,0 +1,167 @@
+		db, err := OpenDatabase(in.dbMetadata, in.dbFilename)
+		if err != nil {
+			t.Errorf("calling OpenDatabase on a new database failed for input %v (error below)", in)
+			t.Fatal(err)
```
why split this into two calls? `t.Fatalf` should be fine

lukechampine commented on persist/boltdb_test.go:
```diff
@@ -0,0 +1,167 @@
+					if err != nil {
+						return err
+					}
+					return nil
```
simplify to `return b.Delete(k)`

lukechampine commented on persist/boltdb_test.go:
```diff
@@ -0,0 +1,167 @@
+	}
+
+	// Generate ten random testInputs for building database files.
+	for i := 0; i < numTestInputs; i++ {
```
when assigning to elements of a slice, use `for i := range slice`

mnsl commented:
Addresses code review comments, except for Luke's note about the weird table driven test format. The randomized testInputs will be replaced with hardcoded ones in a future pull request in order to reimplement via anonymous struct.
